File: benchmarks/oopsla25/subjects/481.py
#!/usr/bin/env python3

import sys

def f481(s: str):
  if s == "":
    return
  i = 0
  while i < len(s)-1:
    if s[i] == "1":
      assert s[i+1] != "1"
    else:
      assert s[i] == "0"
    i += 1
  if len(s) > 1:
    if s[i-1] == "1":
      assert s[i] == "0"
    else:
      assert s[i] == "0" or s[i] == "1"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f481(my_string)

File: benchmarks/oopsla25/subjects/481.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f481(const char *s) {
    if (strlen(s) == 0) {
        return;
    }
    int i = 0;
    while (i < strlen(s) - 1) {
        if (s[i] == '1') {
            assert(s[i + 1] != '1');
        } else {
            assert(s[i] == '0');
        }
        i++;
    }
    if (strlen(s) > 1) {
        if (s[i - 1] == '1') {
            assert(s[i] == '0');
        } else {
            assert(s[i] == '0' || s[i] == '1');
        }
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f481(my_string);
  return 0;
}
